### PR TITLE
Types

### PR DESCRIPTION
**Pull request by lmazuel**

### CODE REVIEW

**Comment by dhawal1304** [comment]
[root@server101 ~]# python3.5
Python 3.5.1 (default, Jul 14 2017, 09:39:18)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import msrest
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/VRTSpython/lib/python3.5/site-packages/msrest/__init__.py", line 27, in <module>
    from .configuration import Configuration
  File "/opt/VRTSpython/lib/python3.5/site-packages/msrest/configuration.py", line 40, in <module>
    from .exceptions import raise_with_traceback
  File "/opt/VRTSpython/lib/python3.5/site-packages/msrest/exceptions.py", line 30, in <module>
    from typing import Callable, Any, Optional, TYPE_CHECKING
ImportError: cannot import name 'TYPE_CHECKING'
>>>
[root@server101 ~]# /opt/VRTSpython/bin/pip3.5  freeze |grep  msrest
msrest==0.5.5
msrestazure==0.5.0

Why i am getting this.
I was trying to install azure modules and after this was trying some azure API.Getting this.

File: msrest/exceptions.py [hunk 1]
```diff
@@ -27,19 +27,28 @@
 import logging
 import sys
 
+from typing import Callable, Any, Optional, TYPE_CHECKING
```
Comment (dhawal1304) [inline msrest/exceptions.py]: This is not working for Python 3.5.1.

[root@server101 ~]# python3.5
Python 3.5.1 (default, Jul 14 2017, 09:39:18)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import msrest
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/VRTSpython/lib/python3.5/site-packages/msrest/__init__.py", line 27, in <module>
    from .configuration import Configuration
  File "/opt/VRTSpython/lib/python3.5/site-packages/msrest/configuration.py", line 40, in <module>
    from .exceptions import raise_with_traceback
  File "/opt/VRTSpython/lib/python3.5/site-packages/msrest/exceptions.py", line 30, in <module>
    from typing import Callable, Any, Optional, TYPE_CHECKING
ImportError: cannot import name 'TYPE_CHECKING'

Comment (dhawal1304) [reply msrest/exceptions.py]: Hi, 
You are right about the version problem. 
But we ship python 3.5.1 with our product as a bundle. So we cannot change the version, we may have to ship a patch of there is no other way

Comment (dhawal1304) [reply msrest/exceptions.py]: Hi, 
You are right about the version problem. 
But we ship python 3.5.1 with our product as a bundle. So we cannot change the version, we may have to ship a patch of there is no other way.

**Comment by lmazuel** [comment]
Hi @dhawal1304 
I don't know exactly your product, but without talking about Azure SDK, why don't you update the bugfix version of your bundled 3.5? There is important security fixes, and security is usually more important than any other considerations. 
See here for 3.5 changelog:
https://docs.python.org/3.5/whatsnew/changelog.html
And you will see how many security issues have been fixed (and then are now publicly known) since 3.5.1

**Comment by dhawal1304** [comment]
Yeah you are right. We are using an ancient python version. Thanks for your insight. What about 3.6.6. Will that work.

**Pull request merged**